### PR TITLE
slack integration part 1

### PR DESCRIPTION
**Pull request by jakemulf**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,22 @@ Once you have your `APP ID` and your tokens, you can fill out the forum at https
 
 It may take a few days for you to get approved.
 
+### Slack
+
+In order to collect data from Slack's API, you need to setup a Slack application. To start you can go to https://api.slack.com/apps/ and click the `Create New App` button. Give your app a name and assign it to your workspace.
```
Comment (margaretkennedy) [inline README.md:36]:
```suggestion
To collect data from Slack's API, you need to set up a Slack application. To start, you can go to https://api.slack.com/apps/ and click the **Create New App** button. Give your app a name and assign it to your workspace.
```

File: README.md [hunk 2]
```diff
@@ -31,6 +31,22 @@ Once you have your `APP ID` and your tokens, you can fill out the forum at https
...
+
+On the next page, click the `Permissions` section underneath `Add features and functionality`. Scroll down to the `Scopes` section and click `Add an OAuth Scope`. This should open up a downdown menu.
```
Comment (margaretkennedy) [inline README.md:38]: my preference is not to use backticks on button names or menu items - bold or plain

Comment (margaretkennedy) [reply README.md:38]:
```suggestion
On the next page, click the `Permissions` section underneath `Add features and functionality`. Scroll down to the `Scopes` section and click `Add an OAuth Scope`. This should open up a dropdown menu.
```

File: README.md [hunk 3]
```diff
@@ -31,6 +31,22 @@ Once you have your `APP ID` and your tokens, you can fill out the forum at https
...
+
+Add the following OAuth scopes: `channels:history`, `channels:read`, `groups:history`, `groups:read`, `im:history`, `im:read`, `mpim:history`, `mpim:read`
+
+Now scroll up to the `OAuth Tokens for Your Workspace` section and click the `Install to Workspace` button. On the next page press `Allow`.
```
Comment (margaretkennedy) [inline README.md:42]:
```suggestion
Now, scroll up to the `OAuth Tokens for Your Workspace` section and click the `Install to Workspace` button. On the next page, press `Allow`.
```

File: README.md [hunk 4]
```diff
@@ -31,6 +31,22 @@ Once you have your `APP ID` and your tokens, you can fill out the forum at https
...
+
+If this was done right, you should now see your app in the `Apps` section of your Slack workspace. Invite the app to any channels you want to track metrics in `/invite @<AppName>`.
```
Comment (margaretkennedy) [inline README.md:44]:
```suggestion
If this was done correctly, you should see your app in your Slack workspace's `Apps` section. Invite the app to any channels you want to track metrics in: `/invite @<AppName>`.
```

File: README.md [hunk 5]
```diff
@@ -31,6 +31,22 @@ Once you have your `APP ID` and your tokens, you can fill out the forum at https
...
+
+Back in the `OAuth Tokens for Your Workspace` section, there should be a `Bot User OAuth Token` field. This is the value used in the `SLACK_API_TOKEN` environmental variable for this project.
+
+Lastly you need your channel IDs. These are simply found in the channel information in your workspace. You can optionally save one of these in the `SLACK_CHANNEL` environmental variable for this project.
```
Comment (margaretkennedy) [inline README.md:48]:
```suggestion
Lastly, you need your channel IDs. These are simply found in the channel information in your workspace. You can optionally save one of these in the `SLACK_CHANNEL` environmental variable for this project.
```

File: README.md [hunk 6]
```diff
@@ -84,9 +99,15 @@ date_increment = to_period("1D")
...
+This example collects data from Slack.
+
+```
+slack = get_channel_messages(SLACK_CHANNEL)
+```
+
 ### Parquet reading and writing
 
-There are two helper methods in `./app.d/parquet_writer.py` that can be used to read and write parquet files, `write_tables` and `read_tables`. These methods work with lists of tables, just like the ones returned by the analytics collectors
+There are two helper methods in `./app.d/parquet_writer.py` that can be used to read and write parquet files, `write_tables` and `read_tables`. `write_tables` expects to receive a list of tables.
```
Comment (margaretkennedy) [inline README.md:110]:
```suggestion
There are two helper methods in `./app.d/parquet_writer.py` that can be used to read and write Parquet files, `write_tables` and `read_tables`. `write_tables` expects to receive a list of tables.
```

File: README.md (hunk 2 above)
Comment (jakemulf) [reply README.md:38]: Cool sounds good, I'll update all of them

**Review by jjbrosnan — APPROVED** [review]

**Pull request merged**